x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DataTypeConverter.java

```diff
@@ -59,9 +60,12 @@ public static DataType commonType(DataType left, DataType right) {
             return left;
         }
         if (isString(left) && isString(right)) {
-            if (left == TEXT) {
+            if (left == TEXT || right == TEXT) {
                 return TEXT;
             }
+            if (left == KEYWORD) {
```

Comment:
I guess the remaining is a combination of `KEYWORD` and `CONSTANT_KEYWORD`. If we want to prioritise a selection, wouldn't a `|| right == KEYWORD` make sense here? If not, couldn't we just `return right;` right away?

Reply:
@bpintea if `left` is `KEYWORD` then return `KEYWORD`. If it's not, the code moves forward and returns `right`. And, at this point in the code, `right` can only be `KEYWORD` or `CONSTANT_KEYWORD`. Adding `|| right == KEYWORD` I think is equivalent and unnecessary? Makes sense or I'm missing something?

Reply:
👍